Staging/master sync (#309)
seqeralabs/nf-tower-docs: docs/wave.md

---
description: 'Using Wave'
---

> Since version 22.4.x Tower supports Seqera's container augmentation service known as Wave also for on-premises installations. 

Author: ```suggestion
From version 22.4.0, Tower supports Seqera's Wave containers service for on-premise installations. 
```

> You can read more about its features [here](https://wave.seqera.io) and its integration with Nextflow [here](https://www.nextflow.io/docs/latest/wave.html)

Author: ```suggestion
See Wave's features [here](https://wave.seqera.io), and Wave integration with Nextflow [here](https://www.nextflow.io/docs/latest/wave.html).
```

## Pairing Tower with Wave

In order to activate the paring mechanism with Wave there are some prerequisites to be fullfilled:

> - https://wave.seqera.io domain should be reachable by Tower network meaning that the domain should be whitelisted in case Tower installation is protected

Author: ```suggestion
- https://wave.seqera.io domain should be accessible to the Tower network (i.e. the domain should be whitelisted in protected Tower installations)
```

- The Tower installation should allow ingress traffic from the Wave service

> - The `TOWER_ENABLE_WAVE` environment variable must be set to `true`

Author: ```suggestion
- The `TOWER_ENABLE_WAVE` environment variable must be set to `true`, or the `enable_wave` parameter set to `true` in your tower.yml configuration
```

> - The tower installation should be available for inbound traffic coming from wave service

Author: ```suggestion
- The tower installation should be available for inbound traffic from the Wave service
```

> - The environment variable `TOWER_ENABLE_WAVE` should be set to the value `true`

Author: ```suggestion
- The `TOWER_ENABLE_WAVE` environment variable should be set to `true`
```

> Once the requisites above are fullfilled the wave functionality will be available for activation directly in Tower UI (feature available only for AWS)

Author: ```suggestion
When these conditions are met, the Wave feature is available on the Tower compute environment creation page (available for AWS compute environments only)
```

> Once wave is enabled a secure channel will be established between Tower installation and Wave service to securely exchange encrypted data.

Author: ```suggestion
Once Wave is enabled, a secure channel is established between Tower and the Wave service to exchange encrypted data securely.
```

> Wave features could also be used by activating wave in the nextflow pipeline config, please read more [here](https://www.nextflow.io/docs/latest/wave.html)

Author: ```suggestion
Wave can also be enabled in the Nextflow pipeline config file. See [here](https://www.nextflow.io/docs/latest/wave.html) for more information.
```
